File: KotlinAlter/src/main/kotlin/com/company/models/Ball.kt
package com.company.models

import java.awt.Color
import java.awt.Graphics
import javax.swing.JComponent

class Ball(internal var xPosition: Int, internal var yPosition: Int): JComponent() {

    fun repaintComponent(x: Int, y: Int){
        this.xPosition = x
        this.yPosition = y
        repaint()
        paintComponent(this.graphics)
    }

    override fun paintComponent(g: Graphics){
        super.paintComponent(g)
        g.color = Color.red
        g.fillOval(this.xPosition, this.yPosition, 50, 50)
    }

}

File: ScalaAlt/src/main/scala/model/Ball.scala
package model

import java.awt.{Color, Graphics}

import javax.swing.JComponent

class Ball(var xPosition: Int,var yPosition: Int) extends JComponent{

  def repaintComponent(x: Int, y: Int): Unit ={
    this.xPosition = x
    this.yPosition = y
    repaint()
    paintComponent(this.getGraphics)
  }

  override def paintComponent(g: Graphics): Unit = {
    super.paintComponent(g)
    g.setColor(Color.red)
    g.fillOval(this.xPosition, this.yPosition, 50, 50)
  }

}
